pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -31,16 +32,43 @@ type StepResolver interface {
 }
 
 type OperatorStepResolver struct {
-	subLister              v1alpha1listers.SubscriptionLister
-	csvLister              v1alpha1listers.ClusterServiceVersionLister
-	client                 versioned.Interface
-	globalCatalogNamespace string
-	resolver               *Resolver
-	log                    logrus.FieldLogger
+	subLister                 v1alpha1listers.SubscriptionLister
+	csvLister                 v1alpha1listers.ClusterServiceVersionLister
+	client                    versioned.Interface
+	globalCatalogNamespace    string
+	resolver                  *Resolver
+	log                       logrus.FieldLogger
+	globalCatalogSourceToggle GlobalNamespaceToggle
 }
 
 var _ StepResolver = &OperatorStepResolver{}
 
+type GlobalNamespaceToggle interface {
+	ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error)
+}
+
+func NewNamespaceAnnotationGlobalCatalogToggle(opClient operatorclient.ClientInterface) *NamespaceAnnotationGlobalCatalogToggle {
+	return &NamespaceAnnotationGlobalCatalogToggle{
+		opClient: opClient,
+	}
+}
+
+type NamespaceAnnotationGlobalCatalogToggle struct {
+	opClient operatorclient.ClientInterface
+}
+
+func (n *NamespaceAnnotationGlobalCatalogToggle) ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error) {
+	fmt.Println("[PING] Getting global catalog source toggle")
+	ns, err := n.opClient.KubernetesInterface().CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+	fmt.Printf("Got namespace %s and error %s", ns, err)
+	if err != nil {
+		return false, err
```

Comment:
In the case of an error fetching the namespace, we should preserve the existing global resolution behavior (which would mean returning true here). It may be more logical to invert the function and return false here instead (to not disable global resolution).